modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurator/ClientExtensionProjectConfigurator.java

```diff
@@ -1161,6 +1162,44 @@ private void _validateClientExtension(
 		}
 	}
 
+	private void _validateGlobalJSScriptElementAttributes(
+		ClientExtension clientExtension) {
+
+		Map<String, Object> typeSettings = clientExtension.typeSettings;
+
+		if (!typeSettings.containsKey("scriptElementAttributes")) {
+			return;
+		}
+
+		Object scriptElementAttributes = typeSettings.get(
+			"scriptElementAttributes");
+
+		if (!(scriptElementAttributes instanceof Map)) {
+			throw new GradleException(
+				"The property 'scriptElementAttributes' must be an object");
+		}
+
+		Map<String, Object> scriptElementAttributesMap =
+			(Map<String, Object>)scriptElementAttributes;
+
+		for (Map.Entry<String, Object> entry :
+				scriptElementAttributesMap.entrySet()) {
+
+			if (Objects.equals(entry.getKey(), "src")) {
+				throw new GradleException(
+					"Use the 'url' property instead of the script attribute " +
+						"'src'");
+			}
+
+			if (entry.getValue() == null) {
+				throw new GradleException(
+					StringBundler.concat(
+						"A value for the attribute '", entry.getKey(),
+						"' must be specified"));
+			}
```

Comment:
What do we do in the case where the attribute value is a non-scalar type (e.g. array or object)?

Reply:
Well, currently it's just translated to a string representation. But tbh I did not think about that. @ethib137 do you want to throw an error or allow it to be translated to string?

e.g.
```
att:
    a: a
    b: b
att2:
    - c
    - d
```

becomes `att="{a=a, b=b}" arr="[c, d]"`

Reply:
Error message suggestion in case we throw an error:
> A scalar non-null value must be defined for the attribute 'attribute-name'

Reply:
Now an error is thrown when a non-scalar value is passed. I also added two new gradle tests to cover these cases.

Please, refer to the [new PR](https://github.com/liferay-platform-experience/liferay-portal/pull/492)